Course Details Page

huangderful opened the pull request:
Created the course details page.

I created a table for the Course Details as well as a Course Details Index page which have tests that 100% pass. 

<img width="1279" alt="image" src="https://github.com/ucsb-cs156-f23/proj-courses-f23-7pm-2/assets/91812464/1b82a637-cbd9-45b5-8aea-7e46523eb538">

Dokku Deployment: https://proj-courses-huangderful-dev-19.dokku-10.cs.ucsb.edu/

Closes #18
Closes #14

JonathanCheng123 reviewed (approved):
LGTM

tigeryu8900 reviewed (approved):
LGTM

GlassBricks commented on frontend/src/main/components/CourseDetails/CourseDetailsTable.js:
```diff
@@ -0,0 +1,63 @@
+  formatTime,
+} from "main/utils/sectionUtils.js";
+
+export default function PersonalSectionsTable({ details }) {
```
Instead of copying `PersonalSectionsTable` with the exact same code, tests, and storybook, you can simply reuse `PersonalSectionsTable`.

In a future PR, you can consider renaming `PersonalSectionsTable` to `CourseDetailsTable` (and maybe related fixtures files) to reflect their more general purpose.

GlassBricks replied:
As a note, in the future we likely will add more information to the course details page, probably not n tabular format. Using the `PersonalSectionsTable` is good enough as a placeholder; Depending on the eventual details page layout it's yet to be decided if we will rename it `CourseDetailsTable`.

pconrad replied:
You should not be exporting default function `PersonalSectionsTable` in a file called `CourseDetailsTable`.   And as @GlassBricks said, if the code is exactly the same, why make a new component?   That's the whole purpose of components: to be reusable pieces of code that you can use to build a user interface.

MuMuKow reviewed (approved):
LGTM

pconrad commented on frontend/package-lock.json:
it may be better to run `npm ci` which doesn't change `package-lock.json`

pconrad commented on frontend/src/main/components/CourseDetails/CourseDetailsTable.js:
```diff
@@ -0,0 +1,63 @@
+  formatTime,
+} from "main/utils/sectionUtils.js";
+
+export default function PersonalSectionsTable({ details }) {
```
At a minimum, this should be changed to match the name of the file

```suggestion
export default function CourseDetailsTable({ details }) {
```

pconrad replied:
But if you take that approach, at a minimum, create a separate refactoring issue that consolidates PersonalSectionsTable and CourseDetailsTable into a single component OR ELSE explains clearly in comments how they are different (other than name and where they are used.)

That's the minimum I'd ask to get this mergeable.  But it would be better to use the existing component.

pconrad reviewed (changes requested):
I agree with @GlassBricks that it doesn't make sense to create a new component that is an exact copy of an already existing component.

Just reuse the existing component, perhaps renaming it to something more generic.

At an absolute minimum, though:
* Rename the one component
* Add a refactoring issue to clear up this technical debt that you may be creating here.

If indeed there is a substantive difference in the tables (other than their name and where they are used), then I take it all back.. but in that case, please outline what that is.

huangderful commented:
> I agree with @GlassBricks that it doesn't make sense to create a new component that is an exact copy of an already existing component.
> 
> Just reuse the existing component, perhaps renaming it to something more generic.
> 
> At an absolute minimum, though:
> 
> * Rename the one component
> * Add a refactoring issue to clear up this technical debt that you may be creating here.
> 
> If indeed there is a substantive difference in the tables (other than their name and where they are used), then I take it all back.. but in that case, please outline what that is.

Alright, I renamed the component and added a refactoring issue.

JonathanCheng123 reviewed (approved):
LGTM on dokku

tigeryu8900 reviewed (approved):
LGTM

JonathanCheng123 reviewed (approved):
LGTM on dokku

pconrad reviewed (approved):
LGTM

raysujf reviewed (approved):
LGTM

pconrad commented on frontend/src/tests/pages/CourseDetails/CourseDetailsIndexPage.test.js:
```diff
@@ -0,0 +1,156 @@
+        params: { qtr: "20221", enrollCode: "06619" },
+      })
+      .reply(200, {
+        quarter: "20221",
```
Agreed with @GlassBricks but we'll let it slide since we are coming up against the deadline.  We can refactor this later.
